File: src/main/java/com/cleancity/backend/device/dto/RegisterDeviceRequest.java
package com.cleancity.backend.device.dto;

import com.cleancity.backend.device.DevicePlatform;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public class RegisterDeviceRequest {
    @NotBlank
    private String deviceId;

    @NotBlank
    private String fcmToken;

    private String deviceName;

    @NotNull
    private DevicePlatform platform;

    public RegisterDeviceRequest() {}

    public String getDeviceId() { return deviceId; }
    public void setDeviceId(String deviceId) { this.deviceId = deviceId; }
    public String getFcmToken() { return fcmToken; }
    public void setFcmToken(String fcmToken) { this.fcmToken = fcmToken; }
    public String getDeviceName() { return deviceName; }
    public void setDeviceName(String deviceName) { this.deviceName = deviceName; }
    public DevicePlatform getPlatform() { return platform; }
    public void setPlatform(DevicePlatform platform) { this.platform = platform; }
}

File: src/main/java/com/cleancity/backend/device/dto/RegisterDeviceRequest.kt
package com.cleancity.backend.device.dto

import com.cleancity.backend.device.DevicePlatform
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class RegisterDeviceRequest(
    @field:NotBlank
    val deviceId: String,

    @field:NotBlank
    val fcmToken: String,

    val deviceName: String? = null,

    @field:NotNull
    val platform: DevicePlatform
)
